### DISCUSSION TITLE
Importing ingredients from a file of some kind

### DISCUSSION THREAD

**Discussion by rosschurchill** [karlomikus/bar-assistant#320] [General]
Hey all,

I made to put together a list of a few thousand ingredients URL which i intend to scrap and then i wanted to import them into BA but i can't seem to work it how the import process looks.

The cocktail import process is great i can get this done via API perfectly but the ingredients is proving tricky

Any advise would be amazing thanks

Regards
R

**Comment by karlomikus** [comment]
Hello.
There is no feature to import ingredients from text currently.
You could use [create ingredient endpoint](https://api.barassistant.app/docs#/operations/efa43043e488624a2f3d8e8123a2d9be) and loop through every line.
Or if you want to be a bit creative with your text editor, you could make SQL insert queries and run the queries on your the database directly.

**Comment by rosschurchill** [comment]
Perfect I missed this in the API docThanks mateOn Aug 28, 2024 5:26 PM, Karlo Mikus ***@***.***> wrote:
Hello.
There is no feature to import ingredients from text currently.
You could use create ingredient endpoint and loop through every line.
Or if you want to be a bit creative with your text editor, you could make SQL insert queries and run the queries on your the database directly.

**Comment by damitjimii** [comment]
In the docker shell for the bar-assistant you can run SQLite to import a csv if you’ve mapped an external folder into your container. 
Chown the file with 33:33 before entering the container shell
> sqlite> .mode csv
You should be in the SQLite shell now
> import <path/filename> tablename

You will likely need to export the table first to be sure you have the correct fields and I’m not sure what the inventory table is, but it should be obvious.
